app/views/comments/_form.html.erb

```diff
@@ -19,16 +19,30 @@
           padding-top: 10px;
         }
       </style>
-
-
-      <%= render :partial => "editor/toolbar" %>
+      <%# if location == reply, toolbar needs reply_to for dropzone IDs %>
+      <% locals = (location == :reply) ? 
+        { :location => location, :reply_to => reply_to } : 
+        { :location => location } 
+      %>
+      <%# location: for creating unique .dropzone IDs %>
+      <%# location can be: edit form, reply form, main comment form, note responses %>
+      <%= render :partial => "editor/toolbar", :locals => locals %>
 
       <% if local_assigns[:reply_to].present? %>
         <%= hidden_field_tag :reply_to, local_assigns[:reply_to] %>
       <% end %>
 
-      <div class="form-group dropzone">
-
+      <%# most comment forms have two dropzones: 1) small button, 2) large form that covers textarea %>
```

Comment:
There's a case possibly to be made that logic doesn't belong in a template, but i see why this may be an exception. But just to explore the possibility, would it be possible to wrap this logic in a function and do it in a single line in the template, with the logic tucked into the `application_helper.rb` (helpers is where template-specific logic often goes, by convention)?

But, i'm not feeling strongly either way, there are pros and cons - it might be more readable in pure Ruby in the helper, but it's also more work to go and find it...

Reply:
Thank you! This is great feedback, and helps me learn more about Rails conventions (which I'm going to be learning for a while). I went ahead and put the logic in `application_helper.rb`, just for the sake of developing good habits.